### PR TITLE
[WIP] batch_norm in numpy

### PR DESCRIPTION
**Pull request by danFromTelAviv**

### CODE REVIEW

**Comment by danFromTelAviv** [comment]
my best guess for what a numpy implementation of batch norm should be. I am not sure exactly what are the expected inputs/ outputs. The tests are a bit confusing as well.

**Comment by danFromTelAviv** [comment]
@gabrieldemarmiesse there you go

**Comment by gabrieldemarmiesse** [comment]
Thanks, can you remove the empty line and the docstrings? I'll start looking at it in a few dozen of minutes. I have another PR on my plate.

**Comment by gabrieldemarmiesse** [comment]
I think the best would be to use it in the tests to make sure it works. Let me see where it can go.

**Comment by gabrieldemarmiesse** [comment]
Since the two batchnorm functions are using three or more input tensors, it would be much easier if #11265 was merged into master. We could use the helper function to test `batch_normalization` and `normalize_batch_in_training` easily.

**Comment by danFromTelAviv** [comment]
sounds good. let me know when it's ready.

**Comment by gabrieldemarmiesse** [comment]
@danFromTelAviv Could you try to have a go at making tests without the helper functions of #11265? You'll need to modify this test: https://github.com/keras-team/keras/blob/5ef0fc3215fb43f5772673b31c51ad8a6582bc39/tests/keras/backend/backend_test.py#L1614 and also create a new test for `batch_normalization`.

**Comment by danFromTelAviv** [comment]
Sure. I'll try to get around to it this weekend. From what I remember when I tried last time I was very confused about what the allowed inputs are. I saw they used None for some of the variables which doesn't make any sense?

**Comment by gabrieldemarmiesse** [comment]
If we have the tests, I'll be able to use a debugger and quickly find any errors/subtle things. We'll also be able to know when it's finished. Ping me when you have tests and I'll get to the bottom of it.

**Comment by danFromTelAviv** [comment]
I'll try to reverse engineer them but it is not clear what the expected inputs are so I am not sure I'll write very good tests...

**Comment by danFromTelAviv** [comment]
I'm having a bit of a problem with cntk. I haven't used it before so I don't really understand how it works. 
Instead of variables, it accepts placeholders? 
How can I fetch its output with multiple outputs, how should I enter variables into it?

**Comment by gabrieldemarmiesse** [comment]
I'm working on this. So you can leave it to me if you want. See https://github.com/gabrieldemarmiesse/keras/tree/numpy_batchnorm

**Comment by danFromTelAviv** [comment]
ok - I'll look for something else to add.

**Comment by danFromTelAviv** [comment]
I see your implementation. that makes more sense.

**Comment by gabrieldemarmiesse** [comment]
See #11556

**Comment by gabrieldemarmiesse** [comment]
I'll close this PR. I suggest we come back to the problem once #11556 is merged.
